### ISSUE TITLE
Incorrect price in back test.

### ISSUE THREAD

**Issue by bizcad** [QuantConnect/Lean#138]
At line 362 in QCAlgorithm.Trading, the following is coded.
<code>var price = security.Price;</code>
The comment right above it says " Set a temporary price for validating market orders."
On 2/9/2010 the price for DNB was 70.72 Close, but QC assigns a price of 82.60 at that point. 
Later on in my algo I catch the OnOrder event and test for it being filled.  When it is filled, both the Order.Price and the OrderEvent.Price are 82.60.

Securities[symbol] shows the correct date 2/9/2010, but it is showing price, low, high all in the 82-84 range with the price being 82.6.  Based upon the volume reported there of 356738, I was able to trace the price 82.60 close for DNB to 20100107 00:00 in the historical data.  As it happens that is the day after my StartDate.

Since this particular algo is a custom data algo, I do not handle OnData(TradeBars data).  It appears that since I do not have an OnData handler, the price is not advancing.

Nick Stein

**Comment by mchandschuh** [comment]
Have you pulled the latest code. I think this is different now. I'm on the road, can check it out later.

**Comment by bizcad** [comment]
I will pull the latest and test.  However, I looked at the repo before I wrote the issue and the file was the same as my local version.

**Comment by bizcad** [comment]
The issue appears to be resolved.  The 82.60 price is still the price in Trading, but the FilledPrice on a orderEvent is now 71.35.  I cannot tell from the daily bars for sure because I am (somewhat arbitrarily) putting the Market order in at 10:30.  The price looks to be in the correct range for the day.

I have a bit more to do and will comment again tomorrow.

**Comment by bizcad** [comment]
The issue is resolved with the github version of Lean.  It is not resolved on the live version on the server.

When I ran the algorithm against the WebGUI, I did not get the transactions expected.  It appears the live version of QC is still running an older version that is giving the incorrect price.  Case in point, my first trade in the trade log shows.

<pre>
Time	Symbol	Price	Quantity	Type	Status	Value	Tag 
2012-10-15T10:00:00	DLTR	16.67331666	1851	Market	4	30862.30914	 
</pre>
the price for the trade showing of 16.67 is the price on 1/7/2010.  

In addition the compiler did not recognize my reference to OrderTicket.  I changed the return from the Buy(...) call to an int which is the old OrderId API.

I am reporting this issue here rather that post in the community.
